### PR TITLE
Create alternate_disjoint_set.py

### PR DESCRIPTION
**Pull request by anubabajide**
This code implements a disjoint set using Lists 
with added heuristics for efficiency
Union by Rank Heuristic and Path Compression

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/disjoint_set/alternate_disjoint_set.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+"""
+This code implements a disjoint set using Lists 
+with added heuristics for efficiency
+Union by Rank Heuristic and Path Compression
+"""
+class DisjointSet:
+    def __init__(self, set_count):
```
Comment (cclauss) [inline data_structures/disjoint_set/alternate_disjoint_set.py:7]: Type hints?  Doctests?  `set_count` is not a single count but is a list or other iterator of counts so `set_counts` (with a trailing s) would be a more intuitive variable name.

File: data_structures/disjoint_set/alternate_disjoint_set.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        else:
+            self.set_count[src_parent] += self.set_count[dst_parent]
+            self.set_count[dst_parent] = 0
+            self.parents[dst_parent] = src_parent
+            joined_set_size = self.set_count[src_parent]
+
+        self.max_set = max(self.max_set, joined_set_size)
+        return True
+
+    def get_parent(self, set):
```
Comment (cclauss) [inline data_structures/disjoint_set/alternate_disjoint_set.py:44]: `set` is a builtin type in Python https://docs.python.org/3.8/library/stdtypes.html#set-types-set-frozenset so it should not be used as a variable name.

**Comment by anubabajide** [comment]
Hey,
I have fixed the issues...
Kindy review

**Comment by cclauss** [comment]
Hey,
You have messed up the [tasklist](https://docs.github.com/en/github/managing-your-work-on-github/about-task-lists) above.

**Comment by anubabajide** [comment]
I've fixed it @cclauss

**Comment by cclauss** [comment]
https://github.com/TheAlgorithms/Python/blob/master/Travis_CI_tests_are_failing.md

**Comment by anubabajide** [comment]
@cclauss I really can't tell why its failing

File: data_structures/disjoint_set/alternate_disjoint_set.py [hunk 3]
```diff
@@ -26,7 +26,6 @@ def merge(self, src: int, dst: int) -> bool:
         True
         >>> A.merge(0, 2)
         True
-        merge already joined sets
         >>> A.merge(0, 1)
```
Comment (cclauss) [inline data_structures/disjoint_set/alternate_disjoint_set.py:29]:
```suggestion
        >>> # merge already joined sets
        >>> A.merge(0, 1)
```

Comment (anubabajide) [reply data_structures/disjoint_set/alternate_disjoint_set.py:29]: I have fixed this already.

**Review by cclauss — APPROVED** [review]

**Comment by anubabajide** [comment]
Finally it passes.
The repo was about 10 MB as you mentioned, but those dependencies 😅😅

**Pull request merged**